### ISSUE TITLE
License Mapping for BSD-3-Clause and BSD-4-Clause

### ISSUE THREAD

**Issue by msymons** [CycloneDX/cyclonedx-core-java#205]
With this addition to license mappings in PR #195 
https://github.com/CycloneDX/cyclonedx-core-java/blob/b664a13f3e7c41a7e086ec508d804c2bd4207140/src/main/resources/license-mapping.json#L67

The consequence is that the component [antlr4](https://github.com/antlr/antlr4) now maps to `BSD-4-Clause` when the intention of the antlr project is the the license should be`BSD-3-Clause`. 

See [LICENSE.txt](https://github.com/antlr/antlr4/blob/master/LICENSE.txt)

From antlr4 POM:

```
    <licenses>
		<license>
			<name>The BSD License</name>
			<url>http://www.antlr.org/license.html</url>
			<distribution>repo</distribution>
		</license>
	</licenses>
```

So,  is this a problem with `cyclonedx-core-java` license mapping,  or is it a problem with `antlr4` POM?

The consequence of thinking that a `BSD-3-Clause` is actually `BSD-4-Clause` is that the latter:
* Adds an "advertising clause" that requires an acknowledgment of the original source in all advertising material. 
* Was rescinded in 1999
* Is not OSI-approved.

**Comment by stevespringett** [comment]
According to wikipedia, "BSD License" is the 4-clause license. Thus CycloneDX-Core-Java is properly identifying the license from the antlr4 pom.

https://en.wikipedia.org/wiki/BSD_licenses

If the antlr4 project is licensed under BSD-3-Clause, then perhaps the project should explicitly state that using the `BSD-3-Clause` SPDX identifier.

**Comment by msymons** [comment]
@stevespringett, the POM for `antl4` say "The BSD License" and not "BSD License".  The wikipedia page says..

> While the original license is sometimes referred to as the "BSD-old", the resulting 3-clause version is sometimes referred to by "BSD-new." Other names include "New BSD", "revised BSD", "BSD-3", or "3-clause BSD". This version has been [vetted](https://en.wikipedia.org/wiki/Vetting) as an Open source license by the OSI as "The BSD License"

ie,  a reading of this is that "The BSD License" is 3-clause and not 4-clause.

So, is the license mapping in cyclonedx-core-java really correct on this point?

Having said that, I will most definitely try to get the `antlr4` project to use the `BSD-3-Clause` SPDX identifier.

**Comment by tmehnert** [comment]
The `antlr4` project is proven to be `BSD-3-Clause`, see [LICENSE.txt](https://github.com/antlr/antlr4/blob/master/LICENSE.txt#L1).

**Comment by msymons** [comment]
> 



> The `antlr4` project is proven to be `BSD-3-Clause`, see [LICENSE.txt](https://github.com/antlr/antlr4/blob/master/LICENSE.txt#L1).

Yes, @tmehnert ,  that's because the license that you link to is the one is based on the PR that I submitted 😄
